cli/tsc.rs

```diff
@@ -525,8 +540,69 @@ impl TsCompiler {
       return Err(ErrBox::from(compile_response.diagnostics));
     }
 
+    self.set_graph_metadata(
+      source_file.url.clone(),
+      &compile_response.emit_map,
+    )?;
     self.cache_emitted_files(compile_response.emit_map)?;
-    ts_compiler.get_compiled_module(&source_file_.url)
+    Ok(())
+  }
+
+  fn get_graph_metadata(&self, url: &Url) -> Option<GraphFileMetadata> {
+    // Try to load cached version:
+    // 1. check if there's 'meta' file
+    let cache_key = self
+      .disk_cache
+      .get_cache_filename_with_extension(url, "graph");
```

Comment:
is `.graph` the new `.meta`?

Reply:
Complementary concept; `.graph` stores list of dependent files for entry point. Both `.graph` and `.meta` are used to invalidate cache TS compiler cache.